### ISSUE TITLE
Condense printing "Waiting for results..."

### ISSUE THREAD

**Issue by jjwallman** [IBM/qiskit-api-py#12]
The current API interface prints "Waiting for results..." multiple times, which can quickly fill the console when running multiple jobs and makes it difficult to interpret the actual progress.

Changing line 406 of IBMQuantumExperience.py to 
print("\rWaiting for results..." + str(_).zfill(4), end= '') 
(the argument to zfill was chosen arbitrarily)
and adding a line
print('\n')
would address this concern but may cause other issues

**Comment by pacomf** [comment]
Hi @jjwallman , you mean when running multiple experiments, no?, if you run a job, the "Waiting for results" string doesn't appear. You can reduce the timeout to wait for the result when you call to the `run_experiment`, and then you can call when you want to the `get_execution method` or `get_result_from_execution` method.

Anyway, we add your proposal to our backlog. :)

**Comment by jjwallman** [comment]
Hi @pacomf, thanks for that, I was using run_experiment but have switched to run_job. I am a little puzzled by the input to run_job though. As far as I can tell, the input has to be a list of dictionaries that each have the key 'qasm'. I would think a dictionary of keys (i.e., experiment labels) and corresponding qasms would be more natural, where ideally the keys would also be used to store the data. Do you know the motivation for the list of dictionary with a (seemingly) redundant key?

Thanks!

**Comment by pacomf** [comment]
yes, we are working in the definition of the API, we'll improve it!

**Comment by jaygambetta** [comment]
Hi @jjwallman this we will change in the future but for now think of qasms a list of quantum circuits and qasm as the circuit. In qiskit-sdk-py we refer to qasms as the quantum program and qasm as a single circuit. You should use run_job whenever possible but try the qiskit-sdk-py and you can build your program there and if we change the backend we plan to make the front end there supported in the future while we improve the full stack. I will but a roadmap online in wiki of qiskit-sdk-py very soon and would love to get your feedback.

**Comment by jjwallman** [comment]
Hi @jaygambetta, thanks for the info and suggestion. I haven't used the sdk yet because I do not fully understand the internal structure of the circuit class, but will look into it in the near future. On that note, is there current functionality to replace gates at specific points in a circuit or find points in a circuit satisfying stated criteria?

**Comment by jaygambetta** [comment]
@jjwallman yes this is but this is more the beta part of the sdk which has an internal DAG which can be used to do these types of things. Maybe we should move this discussion to the sdk but the internal structure should be pretty clear and if not when you have a look let me know what things need better explanation or improvement.
